jraft-core/src/main/java/com/alipay/sofa/jraft/RouteTable.java

```diff
@@ -86,11 +86,7 @@ public boolean updateConfiguration(final String groupId, final Configuration con
     private GroupConf getOrCreateGroupConf(final String groupId) {
         GroupConf gc = this.groupConfTable.get(groupId);
         if (gc == null) {
-            gc = new GroupConf();
-            final GroupConf old = this.groupConfTable.putIfAbsent(groupId, gc);
-            if (old != null) {
-                gc = old;
-            }
+            return this.groupConfTable.computeIfAbsent(groupId, key -> new GroupConf());
```

Comment:
Can we only keep this line as is and delete everything else in this method?

Reply:
> Can we only keep this line as is and delete everything else in this method?

https://bugs.openjdk.java.net/browse/JDK-8161372